lib/application.js

```diff
@@ -241,6 +244,66 @@ app.use = function use(fn) {
 };
 
 /**
+ * Proxy `Router#disuse()` to remove middleware from the app router.
+ * See Router#disuse() documentation for details.
+ *
+ * If the _fn_ parameter is an express app, then it will be
+ * dismounted from the _route_ specified.
+ *
+ * @public
+ */
+
+app.disuse = function use(fn) {
+  var offset = 0;
+  var path = '/';
+
+  // default path to '/'
+  // disambiguate app.use([fn])
+  if (typeof fn !== 'function')  {
+    var arg = fn;
+
+    while (Array.isArray(arg) && arg.length !== 0)  {
+      arg = arg[0];
+    }
+
+    // first arg is the path
+    if (typeof arg !== 'function')  {
+      offset = 1;
+      path = fn;
+    }
+  }
+
+  var fns = flatten(slice.call(arguments, offset));
+
+  if (fns.length === 0)  {
+    throw new TypeError('app.use() requires middleware functions');
+  }
+
+  // setup router
+  this.lazyrouter();
+  var router = this._router;
+
+  fns.forEach(function (fn)  {
+    // non-express app
+    if (!fn || !fn.handle || !fn.set)  {
+      return router.disuse(path, fn);
+    }
+
+    debug('.disuse app under %s', path);
```

Comment:
So when you use an app, we do a bunch of manipulation on it like setting up setting inheritance, req/res protos, etc. I'm not sure I see that being undone here. If not, please add it :)